### PR TITLE
Upload fixes

### PR DESCRIPTION
**Pull request by antgonza**
This solves #599 and partially solves #795. Will finish #795 in another PR.

### CODE REVIEW

**Comment by antgonza** [comment]
This is ready for review: @ElDeveloper @squirrelo @adamrp, thanks!

**Comment by ElDeveloper** [comment]
hehehe 
screen shot 2015-01-18 at 9 53 14 am

File: qiita_pet/handlers/upload.py [hunk 1]
```diff
@@ -110,19 +112,14 @@ def get(self):
...
-        _, fp = get_mountpoint("uploads")[0]
-        tfp = join(fp, study_id,
-                   resumable_filename + '.part.' + resumable_chunk_number)
-
-        if exists(tfp):
-            self.set_status(200)
-        else:
-            self.set_status(400)
+        # in the original version we used to check if a chunk was already
+        # uploaded and if it was we will send self.set_status(200). Now, as
```
Comment (ElDeveloper) [inline qiita_pet/handlers/upload.py]: will -> would

File: qiita_pet/handlers/upload.py [hunk 2]
```diff
@@ -110,19 +112,14 @@ def get(self):
...
+        # we are not chuncking by file we can simply pass the no exists
```
Comment (ElDeveloper) [inline qiita_pet/handlers/upload.py]: chuncking -> chunking ... I realize these are my misspellings :P

File: qiita_db/test/test_util.py [hunk 3]
```diff
@@ -470,11 +470,19 @@ def test_get_mountpoint(self):
...
         exp = ['uploaded_file.txt']
         obs = get_files_from_uploads_folders("1")
         self.assertEqual(obs, exp)
 
+        # something has been uploaded and not ignoring hidden files/folders
+        # and folders
+        exp = ['.hidden_file.txt', 'a_folder', 'uploaded_file.txt']
+        obs = get_files_from_uploads_folders("1",
+                                             ignore_hidden_and_folders=False)
+        self.assertEqual(sorted(obs), sorted(exp))
```
Comment (ElDeveloper) [inline qiita_db/test/test_util.py]: You should use [`assertItemsEqual`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertItemsEqual).

**Comment by ElDeveloper** [comment]
:+1:

**Comment by antgonza** [comment]
Ready for final review/merge!

File: qiita_db/util.py [hunk 4]
```diff
@@ -506,13 +506,15 @@ def compute_checksum(path):
...
 
-def get_files_from_uploads_folders(study_id):
+def get_files_from_uploads_folders(study_id, ignore_hidden_and_folders=True):
     """Retrive files in upload folders
 
     Parameters
     ----------
     study_id : str
         The study id of which to retrive all upload folders
+    ignore_hidden_and_folders : bool
```
Comment (squirrelo) [inline qiita_db/util.py]: Please update as optional with default True.

File: qiita_pet/static/vendor/js/resumable-uploader.js [hunk 5]
```diff
@@ -50,10 +50,10 @@
        this.resumable = new Resumable({
            chunkSize:3*1024*1024,
            maxFileSize:this.maxFileSize*1024*1024*1024,
-           simultaneousUploads: 6,
+           simultaneousUploads: 1,
```
Comment (squirrelo) [inline qiita_pet/static/vendor/js/resumable-uploader.js]: Why the drop in uploads?

File: qiita_db/util.py [hunk 6]
```diff
@@ -506,13 +506,15 @@ def compute_checksum(path):
     return crc & 0xffffffff
 
 
-def get_files_from_uploads_folders(study_id):
+def get_files_from_uploads_folders(study_id, ignore_hidden_and_folders=True):
```
Comment (squirrelo) [inline qiita_db/util.py]: Would it be better to have this as two bools, one for hidden and one for folders? Would be more useful in the long term, especially if we start allowing folder uploads.

Comment (antgonza) [inline qiita_db/util.py]: Tried to do that but the logic was a bit odd and decided to leave it as a single one. Note that we are not using anywhere this functionality and if we actually need it we can figure it out then.

File: qiita_db/util.py (hunk 4 above)
Comment (antgonza) [inline qiita_db/util.py]: done

File: qiita_pet/static/vendor/js/resumable-uploader.js (hunk 5 above)
Comment (antgonza) [inline qiita_pet/static/vendor/js/resumable-uploader.js]: In the past we were saving all chunks which allowed us to do parallel uploads. Now, we save everything in the same file and we need the chucks in a defined order, thus 1 by 1.

File: qiita_db/util.py (hunk 6 above)
Comment (squirrelo) [inline qiita_db/util.py]: Thought we were moving away from that type of logic for programming, but if everyone else agrees, fine.

Comment (antgonza) [inline qiita_db/util.py]: Not sure what you mean, wanna clarify? We could also discuss during
meeting, if easier.

**Comment by antgonza** [comment]
Ready for merge.

**Pull request merged**